Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.mm

```diff
@@ -1551,6 +1561,21 @@ void getSupportedTypes(HashSet<String>& types) const final
 {
     return m_enabledVideoTrackID && *m_enabledVideoTrackID == trackID;
 }
+
+void MediaPlayerPrivateWebM::layerDidReceiveError(AVSampleBufferDisplayLayer*, NSError*)
+{
+    setNetworkState(MediaPlayer::NetworkState::DecodeError);
```

Comment:
We could protect-this ourselves since the caller is not doing it, ditto below and for SourceBufferPrivate.

Reply:
changing the networkstate will not cause the parent to delete us.